File: dart/dartfind/test/file_types_test.dart
import 'package:dartfind/dartfind.dart' show FileTypes, FileType;
import 'package:test/test.dart';

void main() {
  var fileTypes = FileTypes();

  test('test archive file', () async {
    var fileName = 'archive.zip';
    expect(await fileTypes.isArchiveFile(fileName), true);
    expect(await fileTypes.isAudioFile(fileName), false);
    expect(await fileTypes.isBinaryFile(fileName), false);
    expect(await fileTypes.isCodeFile(fileName), false);
    expect(await fileTypes.isFontFile(fileName), false);
    expect(await fileTypes.isImageFile(fileName), false);
    expect(await fileTypes.isTextFile(fileName), false);
    expect(await fileTypes.isVideoFile(fileName), false);
    expect(await fileTypes.isXmlFile(fileName), false);
    expect(await fileTypes.getFileType(fileName), FileType.archive);
  });

  test('test audio file', () async {
    var fileName = 'music.mp3';
    expect(await fileTypes.isArchiveFile(fileName), false);
    expect(await fileTypes.isAudioFile(fileName), true);
    expect(await fileTypes.isBinaryFile(fileName), false);
    expect(await fileTypes.isCodeFile(fileName), false);
    expect(await fileTypes.isFontFile(fileName), false);
    expect(await fileTypes.isImageFile(fileName), false);
    expect(await fileTypes.isTextFile(fileName), false);
    expect(await fileTypes.isVideoFile(fileName), false);
    expect(await fileTypes.isXmlFile(fileName), false);
    expect(await fileTypes.getFileType(fileName), FileType.audio);
  });

  test('test binary file', () async {
    var fileName = 'binary.exe';
    expect(await fileTypes.isArchiveFile(fileName), false);
    expect(await fileTypes.isAudioFile(fileName), false);
    expect(await fileTypes.isBinaryFile(fileName), true);
    expect(await fileTypes.isCodeFile(fileName), false);
    expect(await fileTypes.isFontFile(fileName), false);
    expect(await fileTypes.isImageFile(fileName), false);
    expect(await fileTypes.isTextFile(fileName), false);
    expect(await fileTypes.isVideoFile(fileName), false);
    expect(await fileTypes.isXmlFile(fileName), false);
    expect(await fileTypes.getFileType(fileName), FileType.binary);
  });

  test('test code file', () async {
    var fileName = 'code.dart';
    expect(await fileTypes.isArchiveFile(fileName), false);
    expect(await fileTypes.isAudioFile(fileName), false);
    expect(await fileTypes.isBinaryFile(fileName), false);
    expect(await fileTypes.isCodeFile(fileName), true);
    expect(await fileTypes.isFontFile(fileName), false);
    expect(await fileTypes.isImageFile(fileName), false);
    expect(await fileTypes.isTextFile(fileName), true);
    expect(await fileTypes.isVideoFile(fileName), false);
    expect(await fileTypes.isXmlFile(fileName), false);
    expect(await fileTypes.getFileType(fileName), FileType.code);
  });

  test('test font file', () async {
    var fileName = 'font.ttf';
    expect(await fileTypes.isArchiveFile(fileName), false);
    expect(await fileTypes.isAudioFile(fileName), false);
    expect(await fileTypes.isBinaryFile(fileName), false);
    expect(await fileTypes.isCodeFile(fileName), false);
    expect(await fileTypes.isFontFile(fileName), true);
    expect(await fileTypes.isImageFile(fileName), false);
    expect(await fileTypes.isTextFile(fileName), false);
    expect(await fileTypes.isVideoFile(fileName), false);
    expect(await fileTypes.isXmlFile(fileName), false);
    expect(await fileTypes.getFileType(fileName), FileType.font);
  });

  test('test image file', () async {
    var fileName = 'image.png';
    expect(await fileTypes.isArchiveFile(fileName), false);
    expect(await fileTypes.isAudioFile(fileName), false);
    expect(await fileTypes.isBinaryFile(fileName), false);
    expect(await fileTypes.isCodeFile(fileName), false);
    expect(await fileTypes.isFontFile(fileName), false);
    expect(await fileTypes.isImageFile(fileName), true);
    expect(await fileTypes.isTextFile(fileName), false);
    expect(await fileTypes.isVideoFile(fileName), false);
    expect(await fileTypes.isXmlFile(fileName), false);
    expect(await fileTypes.getFileType(fileName), FileType.image);
  });

  test('test text file', () async {
    var fileName = 'README.md';
    expect(await fileTypes.isArchiveFile(fileName), false);
    expect(await fileTypes.isAudioFile(fileName), false);
    expect(await fileTypes.isBinaryFile(fileName), false);
    expect(await fileTypes.isCodeFile(fileName), false);
    expect(await fileTypes.isFontFile(fileName), false);
    expect(await fileTypes.isImageFile(fileName), false);
    expect(await fileTypes.isTextFile(fileName), true);
    expect(await fileTypes.isVideoFile(fileName), false);
    expect(await fileTypes.isXmlFile(fileName), false);
    expect(await fileTypes.getFileType(fileName), FileType.text);
  });

  test('test video file', () async {
    var fileName = 'movie.mp4';
    expect(await fileTypes.isArchiveFile(fileName), false);
    expect(await fileTypes.isAudioFile(fileName), false);
    expect(await fileTypes.isBinaryFile(fileName), false);
    expect(await fileTypes.isCodeFile(fileName), false);
    expect(await fileTypes.isFontFile(fileName), false);
    expect(await fileTypes.isImageFile(fileName), false);
    expect(await fileTypes.isTextFile(fileName), false);
    expect(await fileTypes.isVideoFile(fileName), true);
    expect(await fileTypes.isXmlFile(fileName), false);
    expect(await fileTypes.getFileType(fileName), FileType.video);
  });

  test('test xml file', () async {
    var fileName = 'markup.xml';
    expect(await fileTypes.isArchiveFile(fileName), false);
    expect(await fileTypes.isAudioFile(fileName), false);
    expect(await fileTypes.isBinaryFile(fileName), false);
    expect(await fileTypes.isCodeFile(fileName), false);
    expect(await fileTypes.isFontFile(fileName), false);
    expect(await fileTypes.isImageFile(fileName), false);
    expect(await fileTypes.isTextFile(fileName), true);
    expect(await fileTypes.isVideoFile(fileName), false);
    expect(await fileTypes.isXmlFile(fileName), true);
    expect(await fileTypes.getFileType(fileName), FileType.xml);
  });

  test('test unknown file', () async {
    var fileName = 'unknown.ZZZ';
    expect(await fileTypes.isArchiveFile(fileName), false);
    expect(await fileTypes.isAudioFile(fileName), false);
    expect(await fileTypes.isBinaryFile(fileName), false);
    expect(await fileTypes.isCodeFile(fileName), false);
    expect(await fileTypes.isFontFile(fileName), false);
    expect(await fileTypes.isImageFile(fileName), false);
    expect(await fileTypes.isTextFile(fileName), false);
    expect(await fileTypes.isVideoFile(fileName), false);
    expect(await fileTypes.isXmlFile(fileName), false);
    expect(await fileTypes.getFileType(fileName), FileType.unknown);
  });
}

File: typescript/tsfind/tests/filetypes.test.ts
/*
 * filetypes.test.js
 *
 * Some tests of filetypes.js
 */

import { FileType } from '../src/filetype';
import { FileTypes } from '../src/filetypes';
import { FindConfig } from '../src/findconfig';

describe('testing filetypes', () => {
  it('testFileTypesArchiveFile', () => {
    const fileTypes = new FileTypes(new FindConfig());
    const filename = 'archive.zip';
    const res: boolean = fileTypes.isArchiveFile(filename);
    expect(res).toBeTruthy();

    const type: FileType = fileTypes.getFileType(filename);
    expect(type).toBe(FileType.Archive);
  });

  it('testFileTypesAudioFile', () => {
    const fileTypes = new FileTypes(new FindConfig());
    const filename = 'music.mp3';
    const res: boolean = fileTypes.isAudioFile(filename);
    expect(res).toBeTruthy();

    const type: FileType = fileTypes.getFileType(filename);
    expect(type).toBe(FileType.Audio);
  });

  it('testFileTypesBinaryFile', () => {
    const fileTypes = new FileTypes(new FindConfig());
    const filename = 'binary.exe';
    const res: boolean = fileTypes.isBinaryFile(filename);
    expect(res).toBeTruthy();

    const type: FileType = fileTypes.getFileType(filename);
    expect(type).toBe(FileType.Binary);
  });

  it('testFileTypesCodeFile', () => {
    const fileTypes = new FileTypes(new FindConfig());
    const filename = 'code.js';
    const res: boolean = fileTypes.isCodeFile(filename);
    expect(res).toBeTruthy();

    const type: FileType = fileTypes.getFileType(filename);
    expect(type).toBe(FileType.Code);
  });

  it('testFileTypesFontFile', () => {
    const fileTypes = new FileTypes(new FindConfig());
    const filename = 'font.ttf';
    const res: boolean = fileTypes.isFontFile(filename);
    expect(res).toBeTruthy();

    const type: FileType = fileTypes.getFileType(filename);
    expect(type).toBe(FileType.Font);
  });

  it('testFileTypesImageFile', () => {
    const fileTypes = new FileTypes(new FindConfig());
    const filename = 'image.png';
    const res: boolean = fileTypes.isImageFile(filename);
    expect(res).toBeTruthy();

    const type: FileType = fileTypes.getFileType(filename);
    expect(type).toBe(FileType.Image);
  });

  it('testFileTypesTextFile', () => {
    const fileTypes = new FileTypes(new FindConfig());
    const filename = 'text.txt';
    const res: boolean = fileTypes.isTextFile(filename);
    expect(res).toBeTruthy();

    const type: FileType = fileTypes.getFileType(filename);
    expect(type).toBe(FileType.Text);
  });

  it('testFileTypesVideoFile', () => {
    const fileTypes = new FileTypes(new FindConfig());
    const filename = 'movie.mp4';
    const res: boolean = fileTypes.isVideoFile(filename);
    expect(res).toBeTruthy();

    const type: FileType = fileTypes.getFileType(filename);
    expect(type).toBe(FileType.Video);
  });

  it('testFileTypesXmlFile', () => {
    const fileTypes = new FileTypes(new FindConfig());
    const filename = 'markup.xml';
    const res: boolean = fileTypes.isXmlFile(filename);
    expect(res).toBeTruthy();

    const type: FileType = fileTypes.getFileType(filename);
    expect(type).toBe(FileType.Xml);
  });

  it('testFileTypesUnknownFile', () => {
    const fileTypes = new FileTypes(new FindConfig());
    const filename = 'unknown.xyz';
    const res: boolean = fileTypes.isUnknownFile(filename);
    expect(res).toBeTruthy();

    const type: FileType = fileTypes.getFileType(filename);
    expect(type).toBe(FileType.Unknown);
  });
});
